app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,134 @@
+import json
+from unittest import TestCase
+
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from core.models import Recipe, Ingredient
+from recipe.serializers import RecipeSerializer
+
+RECIPE_URL = reverse('recipe:recipe-list')
+
+
+def sample_recipe(name: str = "Pizza",
+                  description: str = "Sour dough") -> Recipe:
+    return Recipe.objects.create(name=name, description=description)
+
+
+def sample_ingredient(recipe: Recipe, name: str = "Cucumber") -> Ingredient:
+    return Ingredient.objects.create(name=name, recipe=recipe)
+
+
+def url_for_recipe(recipe_id: str):
+    return reverse('recipe:recipe-detail', args=[recipe_id])
+
+
+class RecipeTests(TestCase):
```

Comment:
You might want to test some edge cases:
- Getting a non-existent recipe by id (should return 404).
- Create a recipe without name, description or ingredients (should this be valid?)
- Create or update the ingredients of a recipe with an empty ingredient name.

Reply:
done!